fuzz/quicly_mock.c

```diff
@@ -0,0 +1,457 @@
+/*
+ * Copyright (c) 2021 Fastly, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include <assert.h>
+#include <malloc.h>
+#include <stdbool.h>
+#include "khash.h"
+#include "quicly.h"
+#include "quicly/sendstate.h"
+#include "quicly/recvstate.h"
+#include "quicly_mock.h"
+
+KHASH_MAP_INIT_INT64(quicly_stream_t, quicly_stream_t *)
+
+mquicly_context_t mquicly_context;
+
+struct st_quicly_conn_t {
+    struct _st_quicly_conn_public_t super;
+    khash_t(quicly_stream_t) * streams;
+};
+
+struct st_quicly_send_context_t {
+};
+
+static quicly_conn_t *create_connection(quicly_context_t *ctx, bool is_client, struct sockaddr *remote_addr,
+                                        struct sockaddr *local_addr)
+{
+    quicly_conn_t *conn = calloc(1, sizeof(*conn));
+    assert(conn != NULL);
+    conn->super.ctx = ctx;
+    if (is_client) {
+        conn->super.local.bidi.next_stream_id = 0;
+        conn->super.local.uni.next_stream_id = 2;
+        conn->super.remote.bidi.next_stream_id = 1;
+        conn->super.remote.uni.next_stream_id = 3;
+    } else {
+        conn->super.local.bidi.next_stream_id = 1;
+        conn->super.local.uni.next_stream_id = 3;
+        conn->super.remote.bidi.next_stream_id = 0;
+        conn->super.remote.uni.next_stream_id = 2;
+    }
+    conn->streams = kh_init(quicly_stream_t);
+
+    conn->super.local.address.sa = *local_addr;
+    conn->super.remote.address.sa = *remote_addr;
+
+    return conn;
+}
+
+int quicly_accept(quicly_conn_t **conn, quicly_context_t *ctx, struct sockaddr *dest_addr, struct sockaddr *src_addr,
+                  quicly_decoded_packet_t *packet, quicly_address_token_plaintext_t *address_token,
+                  const quicly_cid_plaintext_t *new_cid, ptls_handshake_properties_t *handshake_properties)
+{
+    *conn = create_connection(ctx, false, src_addr, dest_addr);
```

Comment:
```suggestion
    *conn = create_connection(ctx, 0, src_addr, dest_addr);
```
I do not mind either ways (I tend to believe that @hfujita is the owner of file), but we might want to use 1 / 0 instead of true / false, as we use that in other places (e.g., the return value of `quicly_is_blocked`), and because it would be better to use the same custom throughout the project.

Reply:
I love consistency. Applied your suggestion in 6ef0c40.